### PR TITLE
Removed unwanted options in currency model and seed file (closes #1611)

### LINKED ISSUE #1611: Get rid of some options in currency model and seed file
1. supports_cash_addr_format - now hardcoded in blockchain client
2. supports_hd_protocol - was used for multiple address generation validation
3. allow_multiple)deposit_addresses - now we allow multiple addresses for all currencies

Should be updated:
- [x] Account #payment_address!
- [x] api_v2/deposits.rb desc should be updated
- [x] Check other files

### PR DESCRIPTION
**Pull request by dinesh-skyach**
Removed unwanted options in currency model and seed file #1611

### CODE REVIEW

File: app/api/api_v2/deposits.rb [hunk 1]
```diff
@@ -45,19 +45,5 @@ class Deposits < Grape::API
         { currency: params[:currency], address: pa.address }
       end
     end
-
-    desc 'Returns new deposit address for account you want to deposit to. ' \
```
Comment (ysv) [inline app/api/api_v2/deposits.rb]: @dinesh-skyach why did you remove this api call 
Now we allow to create multiple payment addresses for all currencies 
You need only to update `desc`.

File: app/models/account.rb [hunk 2]
```diff
@@ -24,22 +24,6 @@ def payment_address
     payment_addresses.last&.enqueue_address_generation || payment_addresses.create!(currency: currency)
   end
 
-  # Attempts to create additional deposit address for account.
```
Comment (ysv) [inline app/models/account.rb]: @dinesh-skyach Don't need to delete this method

File: config/templates/seed/currencies.yml.erb [hunk 3]
```diff
@@ -21,11 +21,6 @@
   quick_withdraw_limit: 0.1
   deposit_fee:          0
   withdraw_fee:         0
-  options:
```
Comment (ysv) [inline config/templates/seed/currencies.yml.erb:24]: Leave empty options

**Review by ysv — CHANGES_REQUESTED** [review]

File: app/api/api_v2/deposits.rb (hunk 1 above)
Comment (dinesh-skyach) [reply app/api/api_v2/deposits.rb]: @ysv this method depends on `allow_multiple_deposit_addresses` and `supports_hd_protocol`. If we are removing these values then, I think this method also needs to be removed.

Comment (ysv) [reply app/api/api_v2/deposits.rb]: yes we this fields were removed because ability to generate multiple address doesn't depend on this values now

**Review by ysv — CHANGES_REQUESTED** [review]

File: app/api/api_v2/deposits.rb [hunk 4]
```diff
@@ -50,7 +50,7 @@ class Deposits < Grape::API
          'The address may be blank because address generation process is still in progress. ' \
          'If this case you should try again later. ' \
          'IMPORTANT: The system may return the same response as GET /api/v2/deposit_address returns in case ' \
```
Comment (ysv) [inline app/api/api_v2/deposits.rb]: Remove everything after `f this case you should try again later.`

**Review by ysv — CHANGES_REQUESTED** [review]

File: config/templates/seed/currencies.yml.erb (hunk 3 above)
Comment (dinesh-skyach) [reply config/templates/seed/currencies.yml.erb:24]: @ysv done

File: app/api/api_v2/deposits.rb (hunk 4 above)
Comment (dinesh-skyach) [reply app/api/api_v2/deposits.rb]: @ysv done

**Review by ysv — APPROVED** [review]

**Pull request merged**